wrapanapi/systems/msazure.py

```diff
@@ -1077,6 +1077,42 @@ def list_compute_images(self):
         return self.resource_client.resources.list(
             filter="resourceType eq 'Microsoft.Compute/images'")
 
+    def list_compute_images_by_resource_group(self, resource_group=None, free_images=None):
+        """
+        Args:
+            resource_group (str): Name of the resource group
+            free_images (bool): Whether to collect image which do not have any resource(VM) linked to it
+        """
+        resource_group = resource_group or self.resource_group
+        image_list = list(
+            self.resource_client.resources.list(
+                filter=f"resourceType eq 'Microsoft.Compute/images' and resourceGroup eq '{resource_group}'"
+            )
+        )
+
+        if not free_images:
+            return image_list
+
+        vm_list = self.list_vms(resource_group=resource_group)
+        if not vm_list:
+            return image_list
+
+        images_used_by_vm = []
+        for vm_name in vm_list:
+            images_used_by_vm.append(
+                self.compute_client.virtual_machines.get(
+                    resource_group_name=resource_group,
+                    vm_name=vm_name
+                ).storage_profile.image_reference.id
+            )
+
+        images_with_no_resources = []
+        for image in image_list:
+            if image.id not in images_used_by_vm:
+                images_with_no_resources.append(image)
+
+        return images_with_no_resources
```

Comment:
We should split this  into two functions, one that just list the images in resource group and then other that returns the list of free images from those list of images!

Reply:
I think list_compute_images()(which already exists) and list_compute_images_by_resource_group()(which I added) are sufficient. I rather not change in  list_compute_images() as it may be used elsewhere and might break their working. 
list_compute_images_by_resource_group() has `resource_group=None, free_images=None` which can be used to achieve what you're mentioning here

Reply:
We had seen some sort pattern in wrapanapi for cloudwash that has a functions specifically to list free/unused resources and not clubbed with any other functionality listing all or listing free.

But not blocking your PR on that !